test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,163 @@
+package scala
+package tools.nsc
+package reporters
+
+import java.io.{ByteArrayOutputStream, StringReader, BufferedReader, PrintStream, PrintWriter}
+import org.junit.Assert._
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.reflect.internal.util._
+
+
+@RunWith(classOf[JUnit4])
+class ConsoleReporterTest {
+  val source = "Test_ConsoleReporter"
+  val batchFile = new BatchSourceFile(source, "For testing".toList)
+  val posWithSource = new OffsetPosition(batchFile, 4)
+  val content = posWithSource.lineContent
+  val writerOut = new ByteArrayOutputStream()
+  val echoWriterOut = new ByteArrayOutputStream()
+  val reporter = createConsoleReporter("s", writerOut, echoWriterOut)
+
+  /** Change maxerrs and maxwarns */
+  reporter.settings.maxerrs.value = 1
+  reporter.settings.maxwarns.value = 1
+
+
+  def createConsoleReporter(inputForReader: String, writerOut: ByteArrayOutputStream, echoWriterOut: ByteArrayOutputStream = null): ConsoleReporter = {
+      val writer = new PrintWriter(writerOut)
+      val reader = new BufferedReader(new StringReader(inputForReader))
+      
+      /** Create reporter with the same writer and echoWriter if None */
+      Option(echoWriterOut) match {
+        case None => new ConsoleReporter(new Settings(), reader, writer)
+        case Some(_) => new ConsoleReporter(new Settings(), reader, writer, new PrintWriter(echoWriterOut))
+      }
```

Comment:
This is an idiom you'll see nowhere. I won't nitpick default to null, but you can just `case null =>` or just `val echo = if (out != null) new PrintWriter(out) else writer`. You might as well name the params `errOut` and `echoOut` since they aren't Writers.